data-sources/prometheus.md

```diff
@@ -1,12 +1,78 @@
+
+## Long Term Storage: Thanos
+
+Prometheus includes a local on-disk time series database (TSDB), but also optionally integrates with remote storage systems. By default, Prometheus stores data to local TSDB which has a retention period of 15 days, any data outside this retention is automatically deleted. In order to store the data for long term, we recommend setting up [Thanos](https://thanos.io/). Thanos is a highly available Prometheus setup with long term storage capabilities and unlimited retention policies. Thanos also helps in downsampling historical data for massive query speedup when querying large time ranges. You can follow [this](https://thanos.io/tip/thanos/getting-started.md/) guide for more details on how to setup Thanos.
+
+## Accessing Prometheus Data
+
+### PromQL
+
+Similar to SQL, Prometheus has a querying language called [PromQL](https://prometheus.io/docs/prometheus/latest/querying/basics/) that can be used to fetch, query and aggregate the time series metrics. The result of a PromQL expression can either be shown as a graph, viewed as tabular data in Prometheus's expression browser, or consumed by external systems via the [HTTP API](https://prometheus.io/docs/prometheus/latest/querying/api/).
+
+PromQL may require some practice before you get the hang of it, but here are some great cheat sheets that can help you get started:
+* https://promlabs.com/promql-cheat-sheet/
+* https://timber.io/blog/promql-for-humans/
+
+Here is an example of a PromQL query that finds the `sum` of all HTTP requests grouped by the label `code` which indicates the status codes of the HTTP request such as `200(Success), 404(Server not Found), 500(Internal Server Error)` etc:
+
+![promql-example](../assets/images/promql-example.png)
+
+### Prometheus API Client
+
+Prometheus metrics can also be programmatically accessed via API client libraries. We have `prometheus-api-client`, a Python wrapper for the Prometheus HTTP API along with additional functionality for further metric processing. This API client can be used to fetch, query and perform metric processing within Python supported environments such as Jupyter notebooks and is particularly useful for Data Scientists who are interested in examining these time series type of data.
+
+* [API Client](https://github.com/AICoE/prometheus-api-client-python)
+
+* [Example Jupyter Notebook](https://github.com/AICoE/prometheus-api-client-python/blob/master/examples/MetricsList_example.ipynb) - This example Jupyter notebook explains the API client library and its functions supported
+
+* [Tutorial](https://learn.openshift.com/ai-machine-learning/prometheus-api-client/) - Interactive tutorial on using the Prometheus API client library
 
 ## Caveats
 
-* beware combanatory explosion of labels
+* As Prometheus metrics have many labels, be careful when combining these labels with other metrics as they can result in an explosion of metric label pairs, making it complex and compute intensive to query
```

Comment:
As Prometheus metrics have many labels, they have to be carefully combined with other metrics...

Reply:
updated